server/service/src/sync/translations/name_store_join.rs

```diff
@@ -77,7 +77,7 @@ impl SyncTranslation for NameStoreJoinTranslation {
         {
             // if the name_store_join is referencing itself, then exclude it
             // this is an invalid configuration which shouldn't be possible.. but is
-            if store.name_id == data.name_id {
+            if store.name_link_id == data.name_id {
```

Comment:
Is this correct? if yes I can't easily see that. For this reason I suggest to use the StoreRepository and then compare `store.name_row.id === data.name_id`